nixos/modules/services/networking/wireguard.nix

```
@@ -124,8 +124,16 @@ let
 
       publicKey = mkOption {
         example = "xTIBA5rboUvnH4htodjb6e697QjLERt1NAB4mZqp8Dg=";
-        type = types.str;
-        description = "The base64 public key the peer.";
+        type = with types; nullOr str;
+        default = null;
+        description = "The base64 public key of the peer.";
+      };
+
+      publicKeyFile = mkOption {
+        example = "/var/wireguard/pubkey";
```

```suggestion
        example = "/var/lib/wireguard/pubkey";
```

https://www.freedesktop.org/software/systemd/man/file-hierarchy.html#/var/lib/

---

I changed this.